typespec-extension/src/main/java/com/azure/autorest/fluent/TypeSpecFluentPlugin.java

```diff
@@ -31,7 +32,7 @@ public class TypeSpecFluentPlugin extends FluentGen {
 
     public TypeSpecFluentPlugin(EmitterOptions emitterOptions, boolean sdkIntegration) {
         super(new TypeSpecPlugin.MockConnection(), "dummy", "dummy");
-
+        SchemaUtilAdapter.adaptForTypeSpec();
```

Comment:
Emm... Given all these, would subclass make more sense (I assume we need subclass for ObjectMapper and ModelMapper)?

Reply:
Yeah, I had the same thought before. 
It seemed a bit overkill to have 2 subclasses only to override two identical methods `isPlainObject()`.
But yeah, they are cleaner and less hacky..

Reply:
Updated. Though in fact, we need to subclass 5 classes: ObjectMapper, ModelPropertyMapper, ModelMapper, FluentObjectMapper and FluentModelMapper..

Alternative is something like the combination: we rely on MapperFactory to return different `PlainObjectChecker` implementations. 
When needing the `PlainObjectChecker`, we do something like `Mappers.getPlainObjectChecker().check()`.